core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -4386,26 +4418,83 @@ protected void validateHavingClause(SqlSelect select) {
     if (having == null) {
       return;
     }
+    //If we have an HAVING clause, the select scope must be an aggregating scope,
+    //so the cast is safe
     final AggregatingScope havingScope =
         (AggregatingScope) getSelectScope(select);
     if (config.conformance().isHavingAlias()) {
-      SqlNode newExpr = expandGroupByOrHavingExpr(having, havingScope, select, true);
+      SqlNode newExpr = expandGroupByOrHavingOrQualifyExpr(having, havingScope, select,
+          true, false);
       if (having != newExpr) {
         having = newExpr;
         select.setHaving(newExpr);
       }
     }
     havingScope.checkAggregateExpr(having, true);
+    // Need to validate the having expression before inferring unknown types, so that any aliases
+    // can be resolved before qualification occurs
+    // TODO: push this change into calcite
```

Comment:
If this is an easy issue to open/submit, I would open this PR.

Reply:
I have it filed as an issue in the current sprint: https://bodo.atlassian.net/browse/BE-3213